### ISSUE TITLE
cursor.each throws "RqlDriverError: No more rows in the cursor"

### ISSUE THREAD

**Issue by bittrance** [rethinkdb/rethinkdb#3826]
The following code throws a no-more-rows exception on an empty table with npmjs.org node driver version 1.16.1. 

    var r = require('rethinkdb')
    r.connect({ host: 'localhost', db: 'test' }).then(function(conn) {
        return r.table('testo').run(conn).then(function(cursor) {
            return cursor.each(function(err, rec) {
                console.log(err)
                console.log(rec)
            })
        }).done()
    })

This is counterintuitive to me; I would expect .each() to iterate zero times on an empty result.

I don't have a system where I can easily downgrade right now, but I think this may be a regression as my test started failing after upgrade from 1.15 to 1.16?

**Comment by danielmewes** [comment]
Thank you for the issue report @bittrance .
I think that is indeed a bug.

At least it doesn't match the documentation http://rethinkdb.com/api/javascript/each/ which says
> If we need to know when the iteration is complete, each also accepts a second onFinished callback.

So I assume that if you only pass in one callback like in your example, it should simply stop calling that once the cursor is exhausted.

Ping @deontologician

**Comment by deontologician** [comment]
Well, you're right it's a regression. I did a bisect and found 580e20 was the culprit, looking to find the cause now

**Comment by danielmewes** [comment]
We should make another JS driver point release once this is merged.

**Comment by deontologician** [comment]
Merged into next and v1.16.x

**Comment by danielmewes** [comment]
@AtnNn can you please push out an update for the JS driver?

**Comment by AtnNn** [comment]
A new version of the JavaScript driver (1.16.2) has been released with this fix.

**Comment by deontologician** [comment]
@bittrance Thanks for reporting this, and for the test, it was really useful in finding the problem quickly.
